### PR TITLE
Re-adds MCR cell swapping

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request
This simply reads MCR cell swapping and all of the relevant pieces of equipment required to enable this. Most of the changes in the PR-removing cells have also been reverted.

Also reverts the annoying change to the company name which is Micron Control Systems.

## Bug fixes
Fixes the cell overlay not showing the correct overlay
Pre-equipped MCRs now actually come pre-equipped

## Changelog
🆑
balance: MCR's are now reloadable once again.
fix: Fixes the MCR cell overlay not showing the correct overlay
fix: Pre-equipped MCRs now actually come pre-equipped
/:cl:

### CODE REVIEW

**Comment by SpaceLoveSs13** [comment]
Keep in mind this time, unlike last time this was reintroduced, you have to do a lot of changes to balance and modular stuff too where they can be bought and stuff and other attachments like self-recharing.

**Comment by Gandalf2k15** [comment]
> Keep in mind this time, unlike last time this was reintroduced, you have to do a lot of changes to balance and modular stuff too where they can be bought and stuff and other attachments like self-recharing.

do you have a list of places I should look?

**Comment by YakumoChen** [comment]
What stops this from just being power creep for MCRs, exactly...?

The damage on them is already pretty healthy now, and I haven't had recent problems with the gun running out of charge too fast at higher tiers (nobody really uses lower tier parts)

**Comment by YakumoChen** [comment]
Looks like you forgot to touch the makeshift cell and enhanced cell

File: modular_skyrat/modules/microfusion/code/microfusion_cell.dm [hunk 1]
```diff
@@ -24,15 +26,15 @@ Essentially, power cells that malfunction if not used in an MCR, and should only
 
 /obj/item/stock_parts/cell/microfusion //Just a standard cell.
 	name = "microfusion cell"
-	desc = "A standard-issue microfusion cell, produced by Micron Control Systems. For safety reasons, they cannot be charged unless they are inside of a compatible Micron Control Systems firearm."
+	desc = "A standard-issue microfusion cell, produced by Micron Control Systems. Smaller than a can of soda, these fulfill the need for a power source where plugging into a recharger is inconvenient or unavailable; although they will eventually run dry due to being shipped without a fuel source."
```
Comment (YakumoChen) [inline modular_skyrat/modules/microfusion/code/microfusion_cell.dm]: This addition doesn't make sense considering it's w_class is much larger than sodas in game

**Comment by YakumoChen** [comment]
The damage buffs are kinda insane power creep too, not gonna lie.
The illegal shotgun is a three shot crit, with none of the downsides of using a shotgun.

**Comment by Gandalf2k15** [comment]
> What stops this from just being power creep for MCRs, exactly...?
> 
> The damage on them is already pretty healthy now, and I haven't had recent problems with the gun running out of charge too fast at higher tiers (nobody really uses lower tier parts)

Well, it's all about trial and error, but this was a fundamental feature that was removed from my gun.

**Comment by YakumoChen** [comment]
> > What stops this from just being power creep for MCRs, exactly...?
> > The damage on them is already pretty healthy now, and I haven't had recent problems with the gun running out of charge too fast at higher tiers (nobody really uses lower tier parts)
> 
> Well, it's all about trial and error, but this was a fundamental feature that was removed from my gun.

I can assume it's because swappable cells give MCR all the benefits of ballistic weaponry with all the benefits of using an energy weapon and almost none of the downsides of either.

Instead of being a tradeoff this is just a huge buff.

**Pull request merged**